src/Calculator/Controls/MathRichEditBox.cpp

```diff
@@ -0,0 +1,51 @@
+#include "pch.h"
+#include "MathRichEditBox.h"
+
+using namespace Platform;
+using namespace CalculatorApp;
+using namespace CalculatorApp::Common;
+using namespace CalculatorApp::Controls;
+using namespace std;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+using namespace Windows::Foundation::Collections;
+
+DEPENDENCY_PROPERTY_INITIALIZATION(MathRichEditBox, MathText);
+
+MathRichEditBox::MathRichEditBox()
+{
+    Windows::ApplicationModel::LimitedAccessFeatures::TryUnlockFeature(
+        "com.microsoft.windows.richeditmath",
+        "H6wflFFz3gkOsAHtG/D9Tg==",
+        "8wekyb3d8bbwe has registered their use of com.microsoft.windows.richeditmath with Microsoft and agrees to the terms of use.");
+
+    TextDocument->SetMathMode(Windows::UI::Text::RichEditMathMode::MathOnly);
+}
+
+String ^ MathRichEditBox::GetMathTextProperty()
+{
+    String ^ text;
+    this->TextDocument->GetMath(&text);
+    return text;
+}
+
+void MathRichEditBox::SetMathTextProperty(String ^ newValue)
+{
+    bool readOnlyState = this->IsReadOnly;
+    this->IsReadOnly = false;
+
+    // Check if this is a MathML string. If it is, then use SetMath so the math text displays properly.
+    wstring value = newValue->Data();
+    if (value.find(L"<mml:math") == 0 || value.find(L"<math") == 0)
```

Comment:
yes it would, i can look for the full header tag

Reply:
That doesn't really fix the issue, let me think on how to solve this problem

Reply:
That would have the same problem I think, does the caller know what type it would be? They could set it. I think a better design might be to just use a TextBlock if it is plain text anyway, would be lighter weight too.